hw/dv/tools/dvsim/vcs.hjson

```diff
@@ -270,6 +270,13 @@
       build_opts: ["-debug_access+all+reverse"]
       run_opts: ["-gui", "-l {run_dir}/simv.log"]
     }
+    {
+      name: vcs_gui_debug
+      // TODO as done for Xcelium see PR #24156
+      is_sim_mode: 1
+      build_opts: ["-debug_access+all+reverse"]
+      run_opts: ["-gui", "-l {run_dir}/simv.log"]
+    }
```

Comment:
If we don't think we have a special "gui debug mode" for VCS, I suggest we just drop this hunk?

Reply:
This will maybe be implemented in the future and it was to have a clearer error message from `util/dvsim/SimCfg.py` file:
```
        # GUI mode is only available for Xcelium for the moment.
        if (self.gui_debug) and (self.tool not in ['xcelium']):
            log.error("GUI debug mode is only available for Xcelium, please remove "
                      "--gui_debug / -gd option or switch to Xcelium tool.")
            sys.exit(1)
```
Do you still reckon to drop it?

Reply:
I think that if you drop the hunk in `vcs.hsjson`, the error message from `SimCfg.py` will still come out? That's probably the right behaviour, I think?

Reply:
No, I am getting this error only:
`ERROR: [modes] Sub mode "vcs_gui_debug" added to mode "gui_debug" was not found!`

Reply:
Oh, I see. I guess the point is that we try to resolve all modes, even ones that aren't actually being used. In which case, maybe this is the right stub!